tests/kernel/workq/work_queue/src/main.c

```diff
@@ -701,12 +725,79 @@ static void test_triggered_wait_expired(void)
 	reset_results();
 }
 
+
+static void msg_provider_thread(void *p1, void *p2, void *p3)
+{
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	msg_t msg;
+
+	k_msgq_put(&triggered_from_msgq_test.msgq, &msg, K_NO_WAIT);
+}
+
+static void triggered_from_msgq_work_handler(struct k_work *work)
+{
+	msg_t msg;
+
+	zassert_true(k_msgq_get(&triggered_from_msgq_test.msgq, &msg, K_NO_WAIT) == 0, NULL);
+}
+
+static void test_triggered_from_msgq_init(void)
```

Comment:
Pedantry, but this is one of those situations where you're testing for a regression that is going to seem really obscure to someone reading this code in the future.  Might be a good idea to explain it here, or just link to the github issue.